src/main/java/frc/robot/Subsystems/ShooterSystem.java

```diff
@@ -44,23 +44,39 @@ public double getPID(){
         double speed = pid.applyAsDouble(getSpeed(),SPEED_POINT);
 
         SmartDashboard.putNumber("Speed", speed);
+        SmartDashboard.putBoolean("isStopped",false);
         return speed;
     }
 
+    public void setPIDStop(){
+        double speed = pid.applyAsDouble(getSpeed(),0);
+
+        SmartDashboard.putNumber("Speed", speed);
+        SmartDashboard.putBoolean("GOT TO HEREEEEEEEEEEEEEEE", true);
+        SmartDashboard.putBoolean("isStopped",true);
+        this.master.set(speed);
+    }
+
+
     public void shoot(){
-        this.master.set(getPID());
+        this.master.set(-getPID());
     }
 
     public void reverse(){
-        this.master.set(-getPID());
+        this.master.set(getPID());
     }
 
     public double getSpeed(){
+        SmartDashboard.putNumber("RPM", this.encoder.getVelocity());
         return this.encoder.getVelocity();
     }
+    public void resetPID(){
+        pid.reset();
+    }
 
     public void stop(){
-        this.master.stopMotor();
+        SmartDashboard.putBoolean("isStopped",true);
+        this.master.set(0.0);
```

Comment:
Delete this line-we're already telling the master to stop

Reply:
Not sure what you mean by this exactly, we are only telling the master to stop once in this function in line 79. And we only use the stop function to stop the actions when the input is ended currently.

Reply:
Once we write master.stopMotor(), it **stops** the motor-but in a smarter way(slower instead of immediately- ie, we want speed 100 m/s to become 0 m/s slowly, otherwise, it'll kill the engine)

Reply:
We tried that with the PID and the StopWheels action already but it didn't exactly work out, I'm not very sure what to do to produce this without using loops which you guys aren't very fond off.

Reply:
Well, the pid didn't work last time, and i think i know why, but i ain't 100 precent sure.
Everytime you use lamda ()-> you need to write it this way;
New PID(time somthing.. , ()-> { return Shuffleboard.getNumber("kp",KP);}, () -> { return ......} );

I may come tommorow to check this option, but other than that, the stopMotor() worked great last time i used it. It is possible tho that something changed.